dxp/architecture/rich_text_editors.md

```diff
@@ -0,0 +1,158 @@
+| Basic Rich Text Features (text formatting, links, images...) | Use **`<liferay-ui:input-editor />`** or the **`<Editor />`** React component
+| Contextual Inline Editing Experience (page-editing, blogs...) | Use **`<liferay-ui:input-editor type="balloon" />`** or the **`<BallonEditor />`** React component |
+
+# APIs
+
+Rich Text Editors in DXP offer a variety of APIs that can be used depending on the context.
+
+## JSP
+
+Traditionally, the most common way to instantiate a Rich Text Editor is through a [JSP](https://github.com/liferay/liferay-portal/search?l=Java+Server+Pages&q=input-editor). There's 2 main JSP APIs you can use:
+
+#### [`liferay-editor:editor`](https://github.com/liferay/liferay-portal/blob/5541574134ec571532683f18408ae3b68c484f3b/modules/apps/frontend-editor/frontend-editor-taglib/src/main/resources/META-INF/liferay-editor.tld#L12-L162)
+
+> Creates an input field for editing rich text
+
+```jsp
+<%@ taglib uri="http://liferay.com/tld/editor" prefix="liferay-editor" %>
+
+<liferay-editor:editor
+	contents="Initial Text"
+	editorName="ckeditor"
+	name="content"
+/>
+```
+
+The provided [`EditorTag`](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/modules/apps/frontend-editor/frontend-editor-taglib/src/main/java/com/liferay/frontend/editor/taglib/servlet/taglib/EditorTag.java) tracks available instances of [`EditorRenderer`](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/modules/apps/frontend-editor/frontend-editor-api/src/main/java/com/liferay/frontend/editor/EditorRenderer.java) and [defers the rendering](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/modules/apps/frontend-editor/frontend-editor-taglib/src/main/java/com/liferay/frontend/editor/taglib/servlet/taglib/EditorTag.java#L65) and initialization of the Rich Text Editors to an existing registered renderer which .
+
+> **Warning:** The tag `liferay-ui:input-editor` was deprecated in `DXP 7.1`. However, many [occurrences](https://github.com/liferay/liferay-portal/search?l=Java+Server+Pages&q=input-editor) might still be found. When possible, **favour the usage of `liferay-editor:editor`** over the deprecated version of the tag
+
+#### [`liferay-editor:resources`](https://github.com/liferay/liferay-portal/blob/5541574134ec571532683f18408ae3b68c484f3b/modules/apps/frontend-editor/frontend-editor-taglib/src/main/resources/META-INF/liferay-editor.tld#L163-L183)
+
+> Loads the necessary scripts and styles of an editor
+
+This tag was initially created to [Provide a new tag to be able to include input-editors Javascript alone (without the HTML)](https://issues.liferay.com/browse/LPS-78451). It allows the developer of an app to include all the necessary scripts that power a Rich Text Editor (CKEditor, TinyMCE, AlloyEditor...) so that their global APIs are later present in the `global` object for a deferred usage.
```

Comment:
Javascript → JavaScript

Fire this "JS engineer" right now!

Also: Provide → provide

Reply:
![yourefired](https://user-images.githubusercontent.com/905006/77949133-c0aee580-72c6-11ea-9882-a6065282638f.gif)

Reply:
For a moment I thought that'd been me... I just realized before turning in my resignation letter:

<img width="345" alt="Screen Shot 2020-04-02 at 16 00 53" src="https://user-images.githubusercontent.com/905006/78258260-3dc2a080-74fb-11ea-97ce-a4227545588e.png">

So, @izaera...

![yourefired](https://user-images.githubusercontent.com/905006/78258292-4d41e980-74fb-11ea-8738-c00b603fd39c.gif)